File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = []
    
    obj = {}
    
    for i in Y:
        if i in obj:
            obj[i] += 1
        else:
            obj[i] = 1
    
    for j in X:
        if j in obj:
            if obj[j] >= 1:
                obj[j] -= 1
                answer.append(j)
    
    answer = sorted(answer, reverse = True)
    
    if len(answer) == 0:
        return "-1"
    elif answer.count("0") == len(answer):
        return "0"
    else:
        return "".join(answer)

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    let answer = "";
    X = X.split("");
    Y = Y.split("");
    const ref = {};
    X.forEach((item) => ref[item] ? ref[item]++ : ref[item] = 1);
    
    Y.forEach((item) => {
        if (ref[item] && ref[item] > 0) {
            ref[item] -= 1;
            answer += item;
        }
    })
    
    if (answer.length === 0) return "-1";
    else if (Number(answer) === 0) return "0";
    return answer.split("").sort((a, b) => b - a).join("");
}